wallet/src/actors/app/handlers/create_data_req.rs

```diff
@@ -25,12 +24,10 @@ pub struct CreateDataReqRequest {
         deserialize_with = "from_generic_type::<_, DataRequestOutputHelper, _>"
     )]
     request: DataRequestOutput,
-    #[serde(
-        serialize_with = "u64_to_string",
-        deserialize_with = "number_from_string"
-    )]
-    fee: u64,
-    fee_type: Option<FeeType>,
+    #[serde(deserialize_with = "deserialize_fee_backwards_compatible")]
+    fee: Fee,
+    #[serde(default)]
+    fee_type: FeeType,
```

Comment:
If the `fee_type` argument is not provided, this should be `None`, not `FeeType::default()`. This way the `fee` field will have priority if it is using the new syntax.

```suggestion
    fee_type: Option<FeeType>,
```

Reply:
Done! That should do it.

Reply:
Cool, seems to be working, except relative should be called weighted.

Reply:
Just did a serde rename for this.